File: leetcode/2583.二叉树中的第 K 大层和.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def kthLargestLevelSum(self, root: Optional[TreeNode], k: int) -> int:
        que = deque([root])
        arr = []
        while que:
            n = len(que)
            tot = 0
            for i in range(n):
                a = que.popleft()
                tot += a.val
                if a.left:
                    que.append(a.left)
                if a.right:
                    que.append(a.right)
            arr.append(tot)
        arr.sort(reverse = True)
        if len(arr) < k:
            return -1
        return arr[k-1]
        

File: leetcode/2583.二叉树中的第 K 大层和.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func kthLargestLevelSum(root *TreeNode, k int) int64 {
    q := []*TreeNode{root}
    ans := []int{}
    for len(q) > 0 {
        n := len(q)
        tot := 0
        for i := 0; i < n; i ++ {
            node := q[0]
            tot += node.Val
            q = q[1:]
            if node.Left != nil {
                q = append(q, node.Left)
            }
            if node.Right != nil {
                q = append(q, node.Right)
            }
        }
        ans = append(ans, tot)
    }
    if len(ans) < k {
        return -1
    }
    sort.Slice(ans, func(i, j int) bool {
        return ans[i] >= ans[j]
    })
    return int64(ans[k - 1])
}